### PR TITLE
Fix some UB in movie_lib.cc

### PR DESCRIPTION
**Pull request by glebm**
Fixes warnings shown by UBSAN during the title movie playback. Mostly misaligned loads and a couple of bad shifts.

This fixes the game crashing on startup on rg350m.

### CODE REVIEW

**Comment by glebm** [comment]
@alexbatalov Can you please take a look at this PR?

**Comment by alexbatalov** [comment]
Thanks for pinging. Will look today.

**Comment by alexbatalov** [comment]
I've refactored your code a bit to match surroundings, hope you don't mind. `movie_lib` needs complete review, so it should not have too much dependencies. Thanks.

**Comment by glebm** [comment]
>Why doesn't getOffset work?

My original `getOffset` looked like this:

```c++
    constexpr auto getOffset = [](uint16_t v) {
        return static_cast<int8_t>(v & 0xFF) + dword_51F018[v >> 8];
    };
```

The refactored one is like this:

```c++
uint8_t getOffset(uint16_t v)
{
    return static_cast<int8_t>(v & 0xFF) + dword_51F018[v >> 8];
}
```

The difference if the return type, that's probably why. It should probably be `int`.

**Comment by giuliano-macedo** [comment]
Hey, I'm pretty sure this PR caused #194

File: src/movie_lib.cc [hunk 1]
```diff
@@ -2794,4 +2792,19 @@ static void _nfPkDecomp(unsigned char* a1, unsigned char* a2, int a3, int a4, in
...
+{
+    return (b[1] << 8) | b[0];
+}
+
+constexpr uint32_t loadUInt32LE(const uint8_t* b)
+{
+    return (b[3] << 24) | (b[2] << 16) | (b[1] << 8) | b[0];
+}
+
+uint8_t getOffset(uint16_t v)
```
Comment (glebm) [inline src/movie_lib.cc:2805]: @alexbatalov As I mentioned in https://github.com/alexbatalov/fallout1-ce/pull/117#issuecomment-1986828872, the return type here (changed from `auto` to `uint8_t`) is incorrect, it should be `int` or `int32_t`

**Pull request merged**